src/Adapter/Category/CommandHandler/AbstractDeleteCategoryHandler.php

```diff
@@ -35,23 +35,35 @@
  */
 abstract class AbstractDeleteCategoryHandler
 {
+    /**
+     * Get products that belong to a category
+     *
+     * @param int $categoryId
+     */
+    protected function getProductsWithoutCategory($categoryId)
```

Comment:
```suggestion
    protected function getProductsWithoutCategory(int $categoryId): array
```

make sure to return empty array here if you don't have any results

Reply:
I think this query belongs to the `Category.php`, not to handler